pkg/volume/gce_pd/gce_util.go

```diff
@@ -149,11 +150,59 @@ func (mounter *gceSafeFormatAndMount) Mount(source string, target string, fstype
 	}
 	args = append(args, options...)
 	args = append(args, source, target)
-	glog.V(5).Infof("exec-ing: /usr/share/google/safe_format_and_mount %v", args)
-	cmd := mounter.runner.Command("/usr/share/google/safe_format_and_mount", args...)
+	safeFormatAndMountPath := "/usr/share/google/safe_format_and_mount"
+	glog.V(5).Infof("exec-ing: %s %v", safeFormatAndMountPath, args)
+	cmd := mounter.runner.Command(safeFormatAndMountPath, args...)
 	dataOut, err := cmd.CombinedOutput()
 	if err != nil {
-		glog.V(5).Infof("error running /usr/share/google/safe_format_and_mount\n%s", string(dataOut))
+		glog.V(5).Infof("error running %s\n%s", safeFormatAndMountPath, string(dataOut))
+		// if safe_format_and_mount is not available on this system use linux utilities directly.
+		if strings.Contains(err.Error(), safeFormatAndMountPath+": no such file or directory") {
+			return mounter.formatAndMount(source, target, fstype, options)
+		}
+	}
+	return err
+}
+
+// Uses unix utils to format and mount the given disk
+func (mounter *gceSafeFormatAndMount) formatAndMount(source string, target string, fstype string, options []string) error {
+
+	// Add the default options from safe_format_and_mount
+	options = append(options, "discard")
+	options = append(options, "defaults")
+
+	// Try to mount the disk. Let 'mount' try to determine the fstype.
```

Comment:
Yeah, if you let mount try to guess the fstype it will return "null" for an unformatted disk. If you supply the fstype mount returns "[...] wrong fs type [...]" meaning that the disk is either unformatted or you supplied the wrong filesystem type.
I have updated the comment to reflect that.